### PR TITLE
Fix ini block member

### PR DESCRIPTION
**Pull request by xezon**

### CODE REVIEW

File: src/game/common/ini/ini.cpp [hunk 1]
```diff
@@ -198,6 +201,8 @@ void INI::Load(Utf8String filename, INILoadType type, Xfer *xfer)
     g_sXfer = xfer;
     Prep_File(filename, type);
 
+    captainslog_dbgassert(!m_endOfFile, "INI::load, EOF at the beginning!");
```
Comment (xezon) [inline src/game/common/ini/ini.cpp:204]: Seen in World Builder.

File: src/game/common/ini/ini.cpp [hunk 2]
```diff
@@ -389,6 +402,10 @@ void INI::Read_Line()
         // Null terminate the buffer
         *cb = '\0';
         ++m_lineNumber;
+
+        captainslog_dbgassert(cb != &m_currentBlock[INI_MAX_CHARS_PER_LINE],
+            "Buffer too small (%d) and was truncated, increase INI_MAX_CHARS_PER_LINE",
+            INI_MAX_CHARS_PER_LINE);
```
Comment (xezon) [inline src/game/common/ini/ini.cpp:404]: Seen in World Builder.

File: src/game/common/ini/ini.h [hunk 3]
```diff
@@ -86,8 +86,8 @@ class INI
 public:
     enum
     {
-        MAX_LINE_LENGTH = 1028,
-        MAX_BUFFER_SIZE = 8192,
+        INI_MAX_CHARS_PER_LINE = 1028,
```
Comment (xezon) [inline src/game/common/ini/ini.h:89]: This name is confirmed from World Builder.

File: src/game/common/ini/ini.h [hunk 4]
```diff
@@ -170,20 +170,22 @@ class INI
...
     File *m_backingFile;
-    char m_buffer[MAX_BUFFER_SIZE];
+    char m_buffer[INI_BUFFER_SIZE];
     int m_bufferReadPos;
     int m_bufferData;
     Utf8String m_fileName;
     INILoadType m_loadType;
     int m_lineNumber;
-    char m_currentBlock[MAX_LINE_LENGTH];
-    char m_blockEnd;
```
Comment (xezon) [inline src/game/common/ini/ini.h:180]: I removed `m_blockEnd` and added one to `m_currentBlock`. This is much better logically.

File: src/game/common/ini/ini.h [hunk 5]
```diff
@@ -170,20 +170,22 @@ class INI
...
+    char m_currentBlock[INI_MAX_CHARS_PER_LINE + 1];
     const char *m_seps;
     const char *m_sepsPercent;
     const char *m_sepsColon;
     const char *m_sepsQuote;
     const char *m_endToken;
     bool m_endOfFile;
+#ifdef GAME_DEBUG_STRUCTS
+    char m_curBlockStart[INI_MAX_CHARS_PER_LINE];
```
Comment (xezon) [inline src/game/common/ini/ini.h:187]: As seen in World Builder. Ideally class sizes match for them.

**Review by jonwil — APPROVED** [review]
Looks good to me. Approved.

**Pull request merged**